ADR/pipeline-service/adr-0001.md

```diff
@@ -0,0 +1,46 @@
+# Service Account configuration for image builds in AppStudio 
+
+Date: Oct 5, 2022
+
+## Status
+
+Proposed
+
+## Context
+
+The `serviceAccount` used for running a `PipelineRun` workload in a workload cluster needs to have the right SCCs and linked credentials to be able to successfully execute a container image build. The linked credentials will mostly originate from the KCP workspace where the `PipelineRun` was synced from.
+Additionally, the SCC association needed for elevating capabilities required for image builds isn't something we'd prefer to expose the user to.
+
+
+
+## Decision
+
+* As a Pipeline Service platform feature, any service/user intending to build a container image would need to create/use the "image-builder" service account in the KCP workspace. 
+Use of any other service accounts will not lead to a successful assembling of the image.
+
+* The AppStudio Build controller would tap into this platform contract to create relevant PipelineRuns for building container images.
+
+## Consequences
+
+
+### Implementation - AppStudio Build Service Controller
+
+* The AppStudio Build controller creates an "image-builder" service account in the KCP workspace/namespace.
+* The AppStudio Build controller links the image pull secret ( previously created by SPI ) with the "image-builder" `serviceAccount` in the KCP workspace/namespace. Upon doing so, the 'updated' "image-builder" `serviceAccount` would get synced into the workload cluster.
```

Comment:
Does this mean that only `AppStudio Build controller` can do a create/patch on the ServiceAccount? 
I ask this because if a user has this capability, they would be able to patch the SA and in that case, what would be the criteria for the controller to access if the SA is good to be synced(by the syncer) to the Workload cluster?